cjwkernel/pandas/types.py

```diff
@@ -644,15 +722,13 @@ class ProcessResult:
     modules are unreachable. Usually that means `error` should be set.
     """
 
-    error: str = ""
-    """Error (if `dataframe` is zero) or warning text."""
+    errors: ProcessResultError = field(default_factory=list)
```

Comment:
This should not be a `moduletype`: it should be a `errors: List[RenderError] = field(default_factory=list)`. `RenderError` should be a dataclass very similar to `atypes.RenderError`.